iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -704,8 +505,14 @@ TYPED_TEST(ServiceDiscovery_test, FindServiceReturnsMaxProducerServices)
 
     // we want to check whether we find all the services, including internal ones like introspection,
     // so we first search for them without creating other services to obtain the internal ones
-    ServiceContainer serviceContainerExp = this->sut.findService(
-        iox::capro::Wildcard, iox::capro::Wildcard, iox::capro::Wildcard, TestFixture::CommunicationKind::KIND);
+
+    ServiceContainer serviceContainerExp;
+    auto findInternalServices = [&](const ServiceDescription& s) { serviceContainerExp.emplace_back(s); };
+    this->sut.findService(iox::capro::Wildcard,
```

Comment:
I think you could use `FerviceDiscoveryPubSub_test::findService()` here as well.

Reply:
I need a specific lambda here with `serviceContainerExp` to collect the internal services.

Reply:
You could use the container from the fixture.

Reply:
No, I need this for comparison ...